### PR TITLE
Update controller to SDK 0.2.1

### PR DESCRIPTION
**Pull request by yorhodes**
- [ ] Need to update to beta3 without ProxiedContract

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,9 +1,9 @@
 {
-  "name": "@abacus-network/apps",
+  "name": "@abacus-network/controller",
   "devDependencies": {
     "@nomiclabs/hardhat-ethers": "^2.0.1",
     "@nomiclabs/hardhat-waffle": "^2.0.1",
-    "@typechain/ethers-v5": "~7.0.0",
+    "@typechain/ethers-v5": "~10.0.0",
```
Comment (jmrossy) [inline package.json]: Why the `~` here when we use a `^` in the monorepo?

File: package.json [hunk 2]
```diff
@@ -18,30 +18,34 @@
...
+    "@abacus-network/hardhat": "^0.2.1-alpha11",
+    "@abacus-network/utils": "^0.2.1-alpha11",
     "@openzeppelin/contracts": "~3.4.2",
     "@openzeppelin/contracts-upgradeable": "~3.4.2",
     "@summa-tx/memview-sol": "^2.0.0"
-  }
+  },
+  "files": [
+    "/dist"
+  ]
```
Comment (jmrossy) [inline package.json]: Is this getting published to npm? If so, I recommend running fixpack on it and adding some missing fields

File: scripts/deploy-controller.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    multiProvider,
+    config,
+    controllerFactories
+  );
+  const contracts = await deployer.deploy();
+
+  console.log(serializeContracts(contracts))
+}
+
+main().then(console.log).catch(console.error);
```
Comment (jmrossy) [inline scripts/deploy-controller.ts]: What does it log on success (in the `then` block) since nothing is returned?

File: src/utils.ts [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import { ethers } from 'ethers';
 
 import { types, utils } from '@abacus-network/utils';
+import { canonizeId } from '@abacus-network/sdk/dist/utils';
```
Comment (jmrossy) [inline src/utils.ts]: Repeating a comment I just left on Nam's PR #537:
The `files` and/or `exports` section of the package.json may not be set correctly if we need to include `dist` in the path here. Should we fix now so it won't break core consumers if we fix later?

**Review by jmrossy — COMMENTED** [review]
Mostly looks good, just a few small things

File: package.json (hunk 1 above)
Comment (nambrot) [reply package.json]: Didn't change it, but updated it now

File: scripts/deploy-controller.ts (hunk 3 above)
Comment (nambrot) [reply scripts/deploy-controller.ts]: It's just been the default line we/I have been copying to run a "top level async function". Removing the then block

File: src/utils.ts (hunk 4 above)
Comment (nambrot) [reply src/utils.ts]: Imported the `utils` package from the root, though i'm personally not a fan of importing these objects, but I also didn't want to do another round of npm publishing just to access the method

**Review by jmrossy — APPROVED** [review]

**Review by jmrossy — APPROVED** [review]

File: src/deploy.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  async deploy() {
+    const contractsMap = await super.deploy();
+
+    // Transfer ownership of routers to controller and recovery manager.
+    await promiseObjAll(
+      objMap(contractsMap, async (local, contracts) => {
+        const router = contracts.router.contract;
+        const config = this.configMap[local];
+        await router.transferOwnership(config.recoveryManager);
+        await router.setController(config.owner);
```
Comment (yorhodes) [inline src/deploy.ts:57]: these probably need to be swapped

Comment (nambrot) [reply src/deploy.ts:57]: Why do they? It's a bit poor naming imo, but owner != controller actually. `transferOwnership` is for the recovery manager, `setController` for the controller. Filed https://github.com/abacus-network/abacus-controller/issues/3

**Pull request merged**